src/transformStateWithClones.js

```diff
@@ -1,13 +1,32 @@
 'use strict';
 
 /**
- * @param {Object} state
- * @param {Object[]} actions
+ * @param {objetoect} state
+ * @param {objetoect[]} actions
  *
- * @return {Object[]}
+ * @return {objetoect[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let objeto = { ...state };
+  const novo = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+        objeto = { ...objeto, ...extraData };
+    } else if (type === 'removeProperties') {
+      for (const key of keysToRemove) {
+        delete objeto[key];
+      }
+    } else {
+      objeto = {};
+    }
+    
+    novo.push({ ...objeto });
```

Comment:
Pushing the copy of the object to the array is correctly placed outside of the conditional blocks, which avoids repetition and conforms to the DRY principle.